tests/ballerina-unit-test/src/test/resources/test-src/statements/transaction/transaction_stmt.bal

```diff
@@ -1,42 +1,60 @@
-public type TrxError record {
-    string message;
-    error? cause;
-    string data;
+public type TrxErrorData record {
+    string message = "";
+    error? cause = ();
+    string data = "";
     !...
 };
 
+public type TrxError error<string, TrxErrorData>;
+
 @final int RETRYCOUNT = 4;
 @final int RETRYCOUNT_2 = -4;
 
 string workerTest = "";
 
 function testTransactionStmt(int i) returns (string) {
     string a = "start";
-    try {
-        transaction {
-            a = a + " inTrx";
-            try {
-                if (i == -1) {
-                    error err = error(" err" );
-                    panic err;
-                } else if (i == 0) {
-                    a = a + " abort";
-                    abort;
-                } else if (i < -1) {
-                    Trxerror err = error(" trxErr", { data: "test" });
-                    panic err;
-                }
-            } catch (TrxError err) {
-                a = a + err.message;
+    var result = trap testTransactionStmtHelper2(a, i);
```

Comment:
No need to use trap or helper functions to test this behavior. Use them only transaction and panic behaviors are tested together. Check all the places.